### DISCUSSION TITLE
Play Music on "current" Device

### DISCUSSION THREAD

**Discussion by MickLesk** [music-assistant/support#2347] [Feature requests and ideas]
Hello!

I have just searched the feature requests and unfortunately found nothing that fits or is similar.

[Edit]: Yes as native, not BrowserMod or anything else. Like Spotify Web

1. i open HomeAssistant via my Windows PC
2. i open MusicAssistant
3. i select a song from my providers
4. i want to play the song “locally”, i.e. on my PC (Chrome) 

The same applies when I open the app (Android) and use MusicAssistant, I want the music to be played on the current device. 

<img width="1718" alt="image" src="https://github.com/music-assistant/support/assets/47820557/a7461753-0cea-4837-8404-2d8bb7df9cc8">

**Comment by OzGav** [comment]
https://music-assistant.io/player-support/snapcast/

**Reply by MrSiO** [reply]
Unfortunetly this does not fill the need expressed by @MickLesk 
There are no Snapcast extention for Chrome on PC.  (Android, Chromecast yes... but not on PC).
"Chrome Web Store" turns out empty on search for "snapcast" or "snapweb"

Jellyfin has this simple option wihout any extention needed (https://github.com/jellyfin/jellyfin). 
Your current browser is part of the device list next to your DNLA and other.

I also with this option would be available.

**Reply by OzGav** [reply]
You don’t install anything on the computr to use this. You just point a browser tab at the sever

**Reply by MrSiO** [reply]
Does it provide the ability to:

- Use your PC Brower
- No additional hardware
- Connect to Music Assistant UI
- Browse songs
- Select a song
- Hit play

Results : Song plays trough your PC


That would be great if snapcast can do that. Can you provide more info on your setup? Thx

**Reply by OzGav** [reply]
Your browser connects to the Snapcast server running on MA and then appears as a player is MA. You then use the MA UI to send whatever music you wish to the browser. If you are using the HA integration then the player will be visible there to use in automations

**Reply by creeg2** [reply]
So there is a way to make this a little more seamless.  There is a [native windows snapclient](https://github.com/stijnvdb88/snap.net) that can run in the taskbar, launch on start to give that zero-user-interaction feel.  Once the desktop client is pointed to the snapserver and MA has the snapcast output, its a one-click operation to get music to play from the MA web UI.  I think this accomplishes all your requirements @MrSiO

**Reply by MickLesk** [reply]
> So there is a way to make this a little more seamless. There is a [native windows snapclient](https://github.com/stijnvdb88/snap.net) that can run in the taskbar, launch on start to give that zero-user-interaction feel. Once the desktop client is pointed to the snapserver and MA has the snapcast output, its a one-click operation to get music to play from the MA web UI. I think this accomplishes all your requirements @MrSiO

Yes, that may all be true, but there's really no need to be so complicated. 
Every other online app also manages to stream directly to the current device. For example Soundcloud, Youtube, Spotify, LastFM and so on. As a normal “user”, I don't want to install any additional software to somehow put together a stream via any service

**Reply by OzGav** [reply]
The information supplied above is what works now. There is a future where playing to the HA companion apps or to a MA app is possible. What is not possible is to say how far away that future is.

**Comment by MrSiO** [comment]
Will try that. Thx.

**Comment by LeoSum8** [comment]
A downside here is that you need to open a separate tab manually. And it is not very intuitive if you don't understand the details of how Music Assistant and Snapcast work and how it is all stitched together.

A ihmo nice solution is provided by Iris, a webfrontend to mopidy. It seems to use snapcast in the background to play music to the browser. Snapweb seems to be integrated into the frontend so that no separate tab is needed. (https://github.com/jaedb/Iris)

**Reply by MickLesk** [reply]
Well, ultimately the best solution is when it works natively without auxiliary programs. e.g. like open.spotify.com, no matter which device I use to call it up, it works. Be it my Nest Hub, my smartphone or my laptop.

**Comment by tomf247** [comment]
I'm not sure if this helps at all, but fwiw, if you have Kodi installed on your PC you can use it as a player. Just make sure DLNA is set up on Kodi. Also, I'm not sure this is required but I'm also allowing web control in the Kodi settings. Then in MA, you select the Kodi device for playback.

**Comment by alanmilinovic** [comment]
This is a must feature if you ask me.

**Comment by tylerlucas** [comment]
I want to +1 this being an important feature that should be prioritized, if possible. Anecdotally, people in my household are resistant to using MA because they have to think about what Player they're going to use prior to opening an App. E.g.: If they are going to play on Sonos, they could open either Spotify or MA. If they are going to play on their phone or laptop, they need to open Spotify, since Snapcast adds a layer of complexity.

This decision tree results in them just using Spotify 100% of the time. However if MA supported playing on local devices natively, they would switch.

**Comment by HeedfulCrayon** [comment]
I'm going to add my +1 as well.  This is an important feature that should be implemented.  I agree with what @tylerlucas says above 100%

**Comment by Jay2645** [comment]
+1 to this. Just installed Music Assistant and tried to play it from my phone through the Home Assistant Mobile App - only to find out that's not as easy as I had hoped!

**Comment by oOStroudyOo** [comment]
+1 please.

**Comment by transcendentphoto** [comment]
+1 please

**Comment by dmitriman** [comment]
+1

**Comment by OzGav** [comment]
Thanks. Doing +1 posts isn’t contributing to the discussion. Just upvote the original post.

**Reply by transcendentphoto** [reply]
Thanks for pointing

**Comment by XAV59213** [comment]
> It's already cool to be able to use it on connected speakers, but it's a shame not to be able to use it remotely using our phones or on the Internet.

**Comment by marcelveldt** [comment]
Please stop with all the +1 comments, it doesn't bring anything. Just use the upvote button.
This request will be implemented when time allows. There are many requests so we choose the priority from the number of votes.
Just a bit of patience, considering the amount of votes (and my own desire to have this) this will be added.

**Comment by OzGav** [comment]
The first iteration of this is in 2.5.0b11

**Reply by BestBackwards** [reply]
Excellent news! Do you have an estimate for when this will make its way into the main release?
Thank you in advance 😄

**Reply by OzGav** [reply]
No it is still in development

**Reply by mwaterbu** [reply]
It's working pretty great in the betas so far! After today's update it is feeling much more polished too.

**Reply by Borgotto** [reply]
Running 2.5.0b24, I've noticed the built-in browser player doesn't allow to go back/skip songs (the buttons are simply grayed out)
Is that not implemented yet or is it bug?

**Reply by OzGav** [reply]
That is working fine here? Try a different browser or go incognito as a test

**Reply by Borgotto** [reply]
Nevermind, cleared the browser's cache and it's back to normal

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
